### ISSUE TITLE
Fix build on evdi git build

### ISSUE THREAD

**Issue by leleobhz** [displaylink-rpm/displaylink-rpm#234]
Hello!

Build evdi from Git is currently broken on Fedora 37 with kernel 6.2.7-200.fc37.x86_64 because both patches in .spec hungs when applied. Build git releases is the only way possible until version [1.13](https://github.com/DisplayLink/evdi/commit/0155c1dcb7956319630fd0a07e66470d459a33f5) got properly released.

Workaround until fixed is install displaylink rpm as notting happened and build evdi separatelly from git repo.

May this 2 patches ( https://github.com/displaylink-rpm/displaylink-rpm/blob/803557fb09e05b7cf3d0f18df98bf54e9296705c/evdi-bundled-devel-patches.diff and https://github.com/displaylink-rpm/displaylink-rpm/blob/803557fb09e05b7cf3d0f18df98bf54e9296705c/evdi-el-fixes.diff ) got updated for new Git updates?

**Comment by elguero** [comment]
Yes, there is a need to update the patches.  I started to do some preliminary work.  Upstream has not released 1.13 and they have not released the DisplayLink driver for 1.13.

**Comment by leleobhz** [comment]
Hello @elguero !

Indeed version 1.13 is not ready, but `make devel` and `make github-release` stopped working. Since this 2 commands intends to work with main branch of evdi, I understand the patch at least for `make devel` must set some tag on rpmbuild to avoid patch application.

What do you think?

**Comment by elguero** [comment]
Hey!

For some time now, it has become a bit complex trying to support Displaylink's way of managing the EVDI repo. They do not have a branch for maintaining current releases.

`make devel` is supposed to pull from the `devel` branch.  That works until they add code that breaks patches or code paths.  This branch is not guaranteed to work with their currently released driver.  The fixes for working on 6.2 are in that branch.  Unfortunately, instead of just adding patches to `devel` to fix things like supporting newer kernels, other patches have been merged in to work with their proprietary driver they are preparing to release.  The driver version has already been increased to 1.13 due to these changes.

`make github-release` is going to pull the latest tarball that has been released (1.12).  That tarball is not going to have any fixes in it for supporting 6.2.  The changes for 6.2 have been mixed in with the other code changes.  Therefore, it may not be that simple to just create a patch for 6.2 to go into 1.12 based on what is in the `devel` branch.  It might be simple; I just have not really investigated yet.  The other thing we must realize is that they could at any time release the new Displaylink driver and then all that time invested into adding 6.2 to 1.12 would be lost.  They do not communicate their timelines.  It is hard to be motivated to do that work only to have them release everything the following day or week (which has happened to me in the past after putting in the time).

We are always caught in a weird state when new kernels are out, and newer versions of Fedora come out.  Displaylink does their development internally and then when they are ready, they push those changes to the Github repo and then we are stuck since we do not have their latest proprietary driver.

Other package maintainers and I have requested a change to their processes to help us maintainers out.  I know that their staff have been trying to make changes to their internal processes.  It just has not happened yet and, unfortunately, seems to get harder and harder.  It ends up taking a lot of time trying to maintain due to current processes.  It used to be simple. 

Anway, I feel for those who are on an OS that does rolling releases when a new kernel shows up.  Displaylink is just not that friendly of a driver to use on those systems.  Even on Enterprise OS, when there is a minor update, it usually requires a patch due to things being backported into their kernels but those are not overly complicated to do compared to the rolling releases when new kernels appear.

After this long explanation, I am not saying it can't or won't be fixed.  Just trying to make it clear that it is not a quick process to do it right.

**Comment by leleobhz** [comment]
Hello @elguero !

Thank you for you explanation. I don't pretend here to ask for ETA and/or force devs for corrections and I'm sorry if my words sounded like this. Also, I appreciate you attention explaining me about how edge versions of evdi works in real world.

I do not know well RPM macros (Despite the fact I know a bit about RPM Packaging), but I guess patch system for release and for git may be handled separatelly when building RPM. If this is possible to be made, may lead to a easier fixing for edge kernel/fedora releases. 

I'll try to check if this kind of handling is possible and if I find a way, I'll PR here too.

**Comment by elguero** [comment]
Hey.  I believe this to be resolved with the new release today.

Please confirm that you are no longer hitting an issue.

**Comment by leleobhz** [comment]
@elguero .spec misses glibc-devel.i686 glibc-devel as BuildRequires:

```
 leonardo  ~  Github  displaylink-rpm  make github-release                                                                                            2   master 
CFLAGS='-m32 -march=i386' \
LDFLAGS='-m32 -march=i386' \
rpmbuild -bb --define "_topdir `pwd`" --define "_sourcedir `pwd`" --define "_rpmdir `pwd`" --define "_specdir `pwd`" --define "_srcrpmdir `pwd`" --define "_buildrootdir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_builddir `mktemp -d /var/tmp/displayportXXXXXX`" --define "_release 1" --define "_daemon_version 5.7.0-61.129" --define "_version 1.13.1" --define "_tmppath `mktemp -d /var/tmp/displayportXXXXXX`" --define "_github 1" displaylink.spec --target=i386
Construindo plataformas de destino: i386
Construindo para o destino i386
configurando SOURCE_DATE_EPOCH=1681948800
Executando (%prep): /bin/sh -e /var/tmp/displayportIuUnfP/rpm-tmp.tah7BF
+ umask 022
+ cd /var/tmp/displayporttknGBq
+ cd /var/tmp/displayporttknGBq
+ rm -rf displaylink-1.13.1
+ /usr/bin/mkdir -p displaylink-1.13.1
+ cd displaylink-1.13.1
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ cd /var/tmp/displayporttknGBq
+ cd displaylink-1.13.1
+ /usr/lib/rpm/rpmuncompress -x /home/leonardo/Github/displaylink-rpm/DisplayLink_USB_Graphics_Software_for_Ubuntu_5.7.0-61.129.zip
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ chmod +x displaylink-driver-5.7.0-61.129.run
+ ./displaylink-driver-5.7.0-61.129.run --noexec --keep
Creating directory displaylink-driver-5.7.0-61.129
Verifying archive integrity...  100%   MD5 checksums are OK. All good.
Uncompressing DisplayLink Linux Driver 5.7.0-61.129  100%  
+ mkdir -p evdi-1.13.1
+ cd /var/tmp/displayporttknGBq
+ cd displaylink-1.13.1
+ /usr/lib/rpm/rpmuncompress -x /home/leonardo/Github/displaylink-rpm/v1.13.1.tar.gz
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ cd evdi-1.13.1
+ echo 'Patch #0 (evdi-el-fixes-1.13.1.diff):'
Patch #0 (evdi-el-fixes-1.13.1.diff):
+ /usr/bin/patch --no-backup-if-mismatch -f -p1 --fuzz=0
patching file module/evdi_drm_drv.h
patching file module/evdi_gem.c
+ sed -i 's/\r//' README.md
+ RPM_EC=0
++ jobs -p
+ exit 0
Executando (%build): /bin/sh -e /var/tmp/displayportIuUnfP/rpm-tmp.W57Ud8
+ umask 022
+ cd /var/tmp/displayporttknGBq
+ CFLAGS='-m32 -march=i386'
+ export CFLAGS
+ CXXFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i386 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection '
+ export CXXFLAGS
+ FFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i386 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -I/usr/lib/gfortran/modules '
+ export FFLAGS
+ FCFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i386 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -I/usr/lib/gfortran/modules '
+ export FCFLAGS
+ VALAFLAGS=-g
+ export VALAFLAGS
+ LDFLAGS='-m32 -march=i386'
+ export LDFLAGS
+ LT_SYS_LIBRARY_PATH=/usr/lib:
+ export LT_SYS_LIBRARY_PATH
+ CC=gcc
+ export CC
+ CXX=g++
+ export CXX
+ cd displaylink-1.13.1
+ cd evdi-1.13.1/library/
+ /usr/bin/make -O -j16 V=1 VERBOSE=1
make[1]: Entering directory '/var/tmp/displayporttknGBq/displaylink-1.13.1/evdi-1.13.1/library'
gcc -I../module -std=gnu99 -fPIC -D_FILE_OFFSET_BITS=64 -m32 -march=i386 $(pkg-config --cflags-only-I libdrm)   -c -o evdi_lib.o evdi_lib.c
In file included from /usr/include/features.h:515,
                 from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdint.h:26,
                 from /usr/lib/gcc/x86_64-redhat-linux/13/include/stdint.h:9,
                 from evdi_lib.c:5:
/usr/include/gnu/stubs.h:7:11: fatal error: gnu/stubs-32.h: No such file or directory
    7 | # include <gnu/stubs-32.h>
      |           ^~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [<builtin>: evdi_lib.o] Error 1
make[1]: Leaving directory '/var/tmp/displayporttknGBq/displaylink-1.13.1/evdi-1.13.1/library'
erro: Status de saída de /var/tmp/displayportIuUnfP/rpm-tmp.W57Ud8 inválido (%build)

RPM build errors:
    Status de saída de /var/tmp/displayportIuUnfP/rpm-tmp.W57Ud8 inválido (%build)
make: *** [Makefile:163: i386/displaylink-1.13.1-1-github_evdi.i386.rpm] Error 1

```

I don't know if this is a bug in spec or Makefile, but since same Make is used for any platform, this may need to be handled someway. Also, since its not directly related, maybe a case for another bug report?
